client/src/components/Markdown/MarkdownDialog.vue

```diff
@@ -1,3 +1,223 @@
+}
+
+type SelectType = "job_id" | "invocation_id" | "history_dataset_id" | "history_dataset_collection_id";
+
+const selectorConfig = {
+    job_id: {
+        labelTitle: "Step",
+    },
+    invocation_id: {
+        labelTitle: "Step",
+    },
+    history_dataset_id: {
+        labelTitle: "Output",
+    },
+    history_dataset_collection_id: {
+        labelTitle: "Output",
+    },
+};
+
+const jobsUrl = `${getAppRoot()}api/jobs`;
+const workflowsUrl = `${getAppRoot()}api/workflows`;
+const invocationsUrl = `${getAppRoot()}api/invocations`;
+const historiesUrl = `${getAppRoot()}api/histories?view=detailed&q=published&qv=True`;
+
+const selectedShow = ref(false);
+const visualizationShow = ref(false);
+const workflowShow = ref(false);
+const historyShow = ref(false);
+const jobShow = ref(false);
+const invocationShow = ref(false);
+const dataShow = ref(false);
+const dataCollectionShow = ref(false);
+const dataHistoryId = ref<string | null>(null as string | null);
+
+const selectedLabelTitle = computed(() => {
+    const config: SelectTitles = selectorConfig[props.argumentType as SelectType] as SelectTitles;
+    return (config && config.labelTitle) || "Select Label";
+});
+
+function getInvocations() {
+    return axios.get(invocationsUrl);
+}
+
+function getJobs() {
+    return axios.get(jobsUrl);
+}
+
+function getWorkflows() {
+    return axios.get(workflowsUrl);
+}
+
+function getHistories() {
+    return axios.get(historiesUrl);
+}
+
+function onData(response: string) {
+    dataShow.value = false;
+    emit("onInsert", `${props.argumentName}(history_dataset_id=${response})`);
+}
+
+interface ObjectReference {
+    id: string;
+}
+
+function onDataCollection(response: ObjectReference) {
+    dataCollectionShow.value = false;
+    emit("onInsert", `${props.argumentName}(history_dataset_collection_id=${response.id})`);
+}
+
+function onJob(response: ObjectReference) {
+    jobShow.value = false;
+    emit("onInsert", `${props.argumentName}(job_id=${response.id})`);
+}
+
+function onInvocation(response: ObjectReference) {
+    invocationShow.value = false;
+    emit("onInsert", `${props.argumentName}(invocation_id=${response.id})`);
+}
+
+function onHistory(response: ObjectReference) {
+    historyShow.value = false;
+    emit("onInsert", `history_link(history_id=${response.id})`);
+}
+
+function onWorkflow(response: ObjectReference) {
+    workflowShow.value = false;
+    emit("onInsert", `${props.argumentName}(workflow_id=${response.id})`);
+}
+
+function onVisualization(response: string) {
+    visualizationShow.value = false;
+    emit("onInsert", response);
+}
+
+function onOk(selectedLabel: string) {
+    selectedLabel = selectedLabel || "<ENTER LABEL>";
+    selectedShow.value = false;
+    if (props.argumentType == "history_dataset_id") {
+        if (props.useLabels) {
+            emit("onInsert", `${props.argumentName}(output="${selectedLabel}")`);
+        } else {
+            getCurrentGalaxyHistory().then((historyId) => {
```

Comment:
I think you can replace all these `getCurrentGalaxyHistory` calls in a reactive way with something like:
```
const { currentUser } = storeToRefs(useUserStore());
const { currentHistoryId } = useUserHistories(currentUser);
```
or directly getting the currentHistoryId from the History Store.

Reply:
I did a change along these lines and I think it really cleaned up things. Thank you.